lib/learning_switch13.rb

```diff
@@ -0,0 +1,26 @@
+# An OpenFlow controller that emulates a layer-2 switch.
+class LearningSwitch < Trema::Controller
+  INGRESS_FRAME_FILTER = 0
+  MAC_LEARNING = 1
+  FORWARDING = 2
+
+  MULTICAST_MAC_ADDR = '01:00:5e:00:00:00'
+  MULTICAST_MAC_ADDR_MASK = 'ff:ff:ff:00:00:00'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.